### ISSUE TITLE
pyo3_runtime.PanicException: assertion failed: self.is_char_boundary(new_len)

### ISSUE THREAD

**Issue by maingoh** [Stranger6667/jsonschema#618]
This schema seem to raise a `PanicException`:
```python3
>>> Draft202012Validator({'type': 'string', 'pattern': 'aaaaaaaèaaéaaaaéè'})
thread '<unnamed>' panicked at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/alloc/src/string.rs:1410:13:
assertion failed: self.is_char_boundary(new_len)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
pyo3_runtime.PanicException: assertion failed: self.is_char_boundary(new_len)
```
It seem more or less related to the accents/non-ascii chars but in some other configuration it works. 

Not working:
```
Draft202012Validator({'type': 'string', 'pattern': 'èééaaaaaaaaaaaaaè'})
```

Working:
```
Draft202012Validator({'type': 'string', 'pattern': 'èééaè'})
```

**Comment by Stranger6667** [comment]
Thank you for reporting this! I am going to take a look at it today

**Comment by Stranger6667** [comment]
The cause is that I've put the limit in **bytes** rather than in characters on the schema representation inside `__repr__`. I wanted to drop it anyway as it is a performance foot gun (the whole schema is serialized to json first). Will issue a new release soon

**Comment by maingoh** [comment]
Thank you @Stranger6667 for the reactivity !

**Comment by Stranger6667** [comment]
Fixed in `0.25.1` (will be released as soon as the build is done)

You are very welcome :)

Please, don't hesitate to reach out if something won't work as expected with `jsonschema` :)
